src/kitovu/sync/plugin/smb.py

```diff
@@ -76,34 +43,57 @@ class SmbPlugin(syncplugin.AbstractSyncPlugin):
 
     def __init__(self) -> None:
         self._connection: SMBConnection = None
-        self._login_info = _LoginInfo()
+        self._info = _ConnectionInfo()
 
-    def connect(self, url: str, options: typing.Dict[str, typing.Any]) -> None:
-        # FIXME custom exception for errors
-        info: _LoginInfo = _parse_url(url)
-        self._login_info = info
-        password = config.get_password(url)
+    def _password_identifier(self) -> str:
```

Comment:
This is needed because the `keyring` library needs some kind of string as an "username"/identifier for a connection. This was the URL before, now the plugin builds a string which is unique for each connection.